src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= c + b || b >= a + c || c >= b + a) {
+      throw new Error(
+        'the longest side is more than the sum of two other sides',
+      );
+    }
+  }
+
+  getArea(): number {
```

Comment:
The `getArea` method should not receive any parameters according to the checklist. All necessary values are already available as object fields.